internal/common/constants.go

```diff
@@ -0,0 +1,163 @@
+	// EnvPodNamespace is the environment variable key for the pod namespace.
+	EnvPodNamespace = "POD_NAMESPACE"
+	// EnvStorageContainer is the environment variable key for the storage container.
+	EnvStorageContainer = "STORAGE_CONTAINER"
+	// EnvSourceStorageContainer is the environment variable key for the source storage container.
+	EnvSourceStorageContainer = "SOURCE_STORAGE_CONTAINER"
+	// EnvAWSApplicationCredentials is the environment variable key for AWS application credentials.
+	EnvAWSApplicationCredentials = "AWS_APPLICATION_CREDENTIALS"
+	// EnvAzureApplicationCredentials is the environment variable key for Azure application credentials.
+	EnvAzureApplicationCredentials = "AZURE_APPLICATION_CREDENTIALS"
+	// EnvGoogleApplicationCredentials is the environment variable key for Google application credentials.
+	EnvGoogleApplicationCredentials = "GOOGLE_APPLICATION_CREDENTIALS"
+	// EnvGoogleStorageAPIEndpoint is the environment variable key for Google storage API endpoint override.
+	EnvGoogleStorageAPIEndpoint = "GOOGLE_STORAGE_API_ENDPOINT"
+	// EnvOpenstackApplicationCredentials is the environment variable key for OpenStack application credentials.
+	EnvOpenstackApplicationCredentials = "OPENSTACK_APPLICATION_CREDENTIALS"
+	// EnvAlicloudApplicationCredentials is the environment variable key for Alicloud application credentials.
+	EnvAlicloudApplicationCredentials = "ALICLOUD_APPLICATION_CREDENTIALS"
+	// EnvOpenshiftApplicationCredentials is the environment variable key for OpenShift application credentials.
+	EnvOpenshiftApplicationCredentials = "OPENSHIFT_APPLICATION_CREDENTIALS"
+	// EnvECSEndpoint is the environment variable key for Dell ECS endpoint.
+	EnvECSEndpoint = "ECS_ENDPOINT"
+	// EnvECSAccessKeyID is the environment variable key for Dell ECS access key ID.
+	EnvECSAccessKeyID = "ECS_ACCESS_KEY_ID"
+	// EnvECSSecretAccessKey is the environment variable key for Dell ECS secret access key.
+	EnvECSSecretAccessKey = "ECS_SECRET_ACCESS_KEY"
+)
+
+// Constants for values to be set against druidv1alpha1.LabelComponentKey
+const (
+	// ClientServiceComponentName is the component name for client service resource.
+	ClientServiceComponentName = "etcd-client-service"
+	// ConfigMapComponentName is the component  name for config map resource.
+	ConfigMapComponentName = "etcd-config"
+	// MemberLeaseComponentName is the component name for member lease resource.
+	MemberLeaseComponentName = "etcd-member-lease"
+	// SnapshotLeaseComponentName is the component name for snapshot lease resource.
+	SnapshotLeaseComponentName = "etcd-snapshot-lease"
+	// PeerServiceComponentName is the component name for peer service resource.
+	PeerServiceComponentName = "etcd-peer-service"
+	// PodDisruptionBudgetComponentName is the component name for pod disruption budget resource.
+	PodDisruptionBudgetComponentName = "etcd-pdb"
+	// RoleComponentName is the component name for role resource.
+	RoleComponentName = "etcd-druid-role"
```

Comment:
In all these constant names, you are not using `Etcd` as prefix where just below you are using `Etcd` as prefix for TLS certs constants.

Reply:
Is there a need to prefix Etcd to each of these resource names? If yes why?

Reply:
because:
i) you are using `etcd-*` prefix in values of all of these constant
ii) just below of defining these const, you are defining some more const where you have used `Etcd` as prefix for all constant, check [here](https://github.com/gardener/etcd-druid/pull/777/files/f790610b049c6c93353f230d03e898ed0d4d02c4#diff-512a5db7451ddb7d0ad28fc0e1b48135b5529270ffe01eb14add4ed0a02746ccR98-R124)
I thought we want to be consistent with naming

Reply:
`etcd-*` names for the component names make sense, because we add them as label values to the resources being deployed. But the constant names themselves need not have the `Etcd` prefix in them, because these are all druid components anyway, and all of them are for etcds anyway. When you're programatically using any of these constants, you would import the package as `druidconstants github.com/gardener/etcd-druid/internal/common` invoke them as `druidconstants.ComponentNameMemberLease`, which is good enough, rather than `druidconstants.EtcdComponentNameMemberLease` or `druidconstants.ComponentNameEtcdMemberLease`. WDYT?

Reply:
make sense, but my question is about consistency in naming convention. eg:
```go
	// ComponentNameCompactionJob is the component name for compaction job resource.
	ComponentNameCompactionJob = "etcd-compaction-job"
	// ComponentNameEtcdCopyBackupsJob is the component name for copy-backup task resource.
	ComponentNameEtcdCopyBackupsJob = "etcd-copy-backups-job"
```
you are using Etcd in copy backup job but not in compaction job , why ?

Reply:
`EtcdCopyBackupsTask` is a well-defined resource type. On the other hand, `EtcdCompaction` refers to the compaction running on the etcd DB, and not `snapshot compaction`. So naming this `ComponentNameEtcdCompactionJob` would not be right, and would be misleading. Hence, We retain this `ComponentNameCompactionJob`, while `ComponentNameEtcdCopyBackupsJob` needs to have `Etcd` in it for obvious reasons, since it refers to the `EtcdCopyBackupsTask` itself.

Reply:
> On the other hand, EtcdCompaction refers to the compaction running on the etcd DB, and not snapshot compaction

what ? please check this comment again https://github.com/gardener/etcd-druid/pull/777#discussion_r1602589672, it is saying `compaction-job` .... etcd-compaction don't run as a job , only snapshot compaction run as a job

Reply:
I think to avoid this confusion in the names, we should simply name this as `ComponentNameSnapshotCompactionJob = "etcd-snapshot-compaction-job"`. Removes all ambiguity. @ishan16696 WDYT?

Reply:
make sense

Reply:
Addressed in https://github.com/gardener/etcd-druid/pull/777/commits/41fb0096497160bbf4e72e1ee225f9089168d3f4. Resolving.